forte/data/data_store.py

```diff
@@ -226,6 +230,217 @@ def __init__(
         """
         self.__deletion_count: dict = {}
 
+    def __getstate__(self):
+        r"""
+        In serialization,
+            1) will serialize the annotation sorted list as a normal list;
+            2) will remove `_onto_file_path`, `_dynamically_add_type`,
+                `tid_ref_dict`, `tid_idx_dict` and `deletion_count` to save space.
+        """
+        state = super().__getstate__()
+        for k in state["_DataStore__elements"]:
+            # build the full `_type_attributes`
+            self._get_type_info(k)
+            state["_DataStore__elements"][k] = list(
+                state["_DataStore__elements"][k]
+            )
+        state.pop("_onto_file_path")
+        state.pop("_dynamically_add_type")
+        state.pop("_DataStore__tid_ref_dict")
+        state.pop("_DataStore__tid_idx_dict")
+        state.pop("_DataStore__deletion_count")
+        state["entries"] = state.pop("_DataStore__elements")
+        state["fields"] = self._type_attributes
+        for _, v in state["fields"].items():
+            v.pop("parent_entry")
+        return state
+
+    def __setstate__(self, state):
+        r"""
+        In deserialization, we
+            1) transform the annotation list back to a sorted list;
+            2) recreate the `tid_ref_dict` and `tid_idx_dict` from `__elements`;
+            3) reset the `deletion_count`.
+        """
+        self.__dict__.update(state)
+
+        self.__elements = self.__dict__.pop("entries")
+        self._type_attributes = self.__dict__.pop("fields", {})
+        self._DataStore__tid_ref_dict = {}
+        self._DataStore__tid_idx_dict = {}
+        self._DataStore__deletion_count = {}
+
+        reset_index = {}
```

Comment:
Please document your logic here, what is `reset_index` doing

Reply:
yes

Reply:
please add comments to the whole code block, it is still a bit mysterious